src/index.js

```diff
@@ -1,11 +1,35 @@
-import React from 'react';
-import PropTypes from 'prop-types';
-import MapboxDraw from '@mapbox/mapbox-gl-draw/dist/mapbox-gl-draw';
+import React from "react";
+import PropTypes from "prop-types";
+import MapboxDraw from "@mapbox/mapbox-gl-draw/dist/mapbox-gl-draw";
 
 export default class DrawControl extends React.Component {
   componentWillMount() {
-    this.draw = new MapboxDraw(this.props);
-    this.context.map.addControl(this.draw);
+    //if custom modes, add to Mapbox draw modes
+    const modes = MapboxDraw.modes;
+    if (this.props.modes) {
+      for (const mode in this.props.modes) {
+        modes[mode] = this.props.modes[mode];
+      }
+    }
+    const mapboxDrawProps = {
+      ...this.props,
+      modes: modes
+    };
+    this.draw = new MapboxDraw(mapboxDrawProps);
+    this.context.map.addControl(this.draw, this.props.position);
+    //hook draw events
+    this.context.map.on("draw.create", this.props.onDrawCreate);
+    this.context.map.on("draw.delete", this.props.onDrawDelete);
+    this.context.map.on("draw.update", this.props.onDrawUpdate);
+    this.context.map.on("draw.combine", this.props.onDrawCombine);
+    this.context.map.on("draw.uncombine", this.props.onDrawUncombine);
```

Comment:
Really small detail. There's a lot of `this.context.map` here, might be cleaner to do a `const { map } = this.context`. Idem for this.props.

Reply:
okay..done